src/common/os_input_output.rs

```diff
@@ -268,10 +268,23 @@ impl OsApi for OsInputOutput {
     fn load_palette(&self) -> Palette {
         let palette = match Colors::new("xresources") {
             Some(palette) => {
-                let fg = hex_to_rgb(&palette.fg);
-                let bg = hex_to_rgb(&palette.bg);
-                let colors: Vec<(u8, u8, u8)> =
-                    palette.colors.iter().map(|c| hex_to_rgb(c)).collect();
+                let fg = if let Some(foreground) = hex_to_rgb(&palette.fg) {
+                    foreground
+                } else {
+                    return default_palette();
+                };
+
+                let bg = if let Some(background) = hex_to_rgb(&palette.bg) {
+                    background
+                } else {
+                    return default_palette();
+                };
+
+                let colors: Vec<(u8, u8, u8)> = palette
+                    .colors
+                    .iter()
+                    .map(|c| hex_to_rgb(c).unwrap())
```

Comment:
:thinking: In my environment, all elements of `palette.colors` are `None`, so it might be better to consider the case `c` is `None`.

Reply:
It might be ugly but practical to use `for` instead of `iter()`. This will get fallback to the default when any item of `palette.colors` is `None`.

```rust
let mut colors: [(u8, u8, u8); 16] = [(0, 0, 0); 16]; // NOTE: `16` is the same as the length of `palette.colors`.
for (idx, color) in palette.colors.iter().enumerate() {
    if let Some(c) = color {
        colors[idx] = hex_to_rgb(c);
    } else {
        return default_palette();
    }
}
```

Reply:
Its not as nice as the `iter()`, but much nicer and more concise as I came up with, since we sadly couldn't use a closure here.